source/frontend/src/components/common/mapFSM/MapStateMachineContext.tsx

```diff
@@ -197,6 +197,15 @@ export const MapStateMachineProvider: React.FC<React.PropsWithChildren<unknown>>
         } else if (geoFilter?.HISTORICAL_FILE_NUMBER_STR) {
           geoFilter.forceExactMatch = false;
           return mapSearch.searchByHistorical(geoFilter);
+        } else if (
```

Comment:
nit: this is starting to get complex. Adding a filter type to the search criteria would help simplify this